compiler-rs/clients_schema_to_openapi/src/paths.rs

```diff
@@ -325,4 +361,28 @@ mod tests {
         assert_eq!(get_path_parameters("{index}{id/"), vec! {"index"});
         assert_eq!(get_path_parameters("{index{id}/"), vec! {"index{id"});
     }
+
+    #[test]
+    fn test_split_summary_desc() {
+        assert_eq!(split_summary_desc("One sentence."),
+                   SplitDesc{
+                       summary: Some(String::from("One sentence")),
+                       description: None
+                   });
+        assert_eq!(split_summary_desc("This is\nstill one. sentence: all; together"),
+                   SplitDesc{
+                       summary: Some(String::from("This is still one. sentence: all; together")),
+                       description: None
+                   });
+        assert_eq!(split_summary_desc("These are two totally. Separate sentences!"),
+                   SplitDesc{
+                       summary: Some(String::from("These are two totally")),
+                       description: Some(String::from("Separate sentences!"))
+                   });
+        assert_eq!(split_summary_desc(""),
+                   SplitDesc{
+                       summary: None,
+                       description: None
+                   });
```

Comment:
There's a pathological case that isn't handled (I've seen a few of them IIRC):

```rust
assert_eq!(split_summary_desc("These are two totally\n\nSeparate sentences!"),
    SplitDesc {
        summary: Some(String::from("These are two totally")),
        description: Some(String::from("Separate sentences!"))
    }
);
```

This can be fixed by using `let desc_no_newlines = desc.replace("\n\n",".\n").replace('\n'," ");`